CorsixTH/Lua/game_ui.lua

```diff
@@ -784,9 +785,23 @@ function GameUI:playAnnouncement(name, priority, played_callback, played_callbac
   self.announcer:playAnnouncement(name, priority, played_callback, played_callback_delay)
 end
 
+-- Get the configured mouse drag button
+function GameUI:_isMouseScrollButtonDown()
+  local mouse_scroll_button
+  if self.app.config.right_mouse_scrolling then
+    mouse_scroll_button = self.buttons_down.mouse_right
+  else
+    mouse_scroll_button = self.buttons_down.mouse_middle
+  end
+  return mouse_scroll_button
+end
+
 function GameUI:onTick()
   local repaint = UI.onTick(self)
-  if not self.buttons_down.mouse_middle then
+
+  
+
+  if not self:_isMouseScrollButtonDown() then
```

Comment:
```suggestion
  if not self:_isMouseScrollButtonDown() then
```
Luacheck error is here, and we don't need three line spaces imo. Maybe one if thought necessary.